updating older versioning api labels to specify patching (#2420)
temporalio/documentation: docs-src/typescript/patching.md

---
id: patching
title: How to patch Workflow code in TypeScript

> description: The TypeScript SDK Patching API let you change Workflow Definitions without causing non-deterministic behavior in current long-running Workflows.

Reviewer: ```suggestion
description: The TypeScript SDK Patching API lets you change Workflow Definitions without causing non-deterministic behavior in current long-running Workflows.
```

sidebar_label: Patching
tags:
  - typescript
  - workflow code
  - patching
  - versioning
---

> Patching APIs let you change Workflow Definitions without causing [non-deterministic behavior](/concepts/what-is-a-workflow-definition#non-deterministic-change) in current long-running Workflows.

Reviewer: ```suggestion
The TypeScript SDK Patching API lets you change Workflow Definitions without causing [non-deterministic behavior](/concepts/what-is-a-workflow-definition#non-deterministic-change) in current long-running Workflows.
```

</details>

### TypeScript SDK Patching API

> In principle, the TypeScript SDK's patching mechanism works in a similar "feature-flag" fashion to the other SDK's, however, the "versioning" API has been updated to a notion of "patching in" code.

Reviewer: ```suggestion
In principle, the TypeScript SDK's patching mechanism works in a similar "feature-flag" fashion to the other SDKs; however, the "versioning" API has been updated to a notion of "patching in" code.
```

<!--SNIPEND-->

#### Step 2: Deprecate patch

> Once we know that all Workflows started with `v1` code have completed we can [deprecate the patch](https://typescript.temporal.io/api/namespaces/workflow#deprecatepatch).

Reviewer: ```suggestion
When we know that all Workflows started with `v1` code have completed, we can [deprecate the patch](https://typescript.temporal.io/api/namespaces/workflow#deprecatepatch).
```

> Deprecated patches bridge between `v2` and `vFinal` (the end result), they work similarly to regular patches by recording a marker in the Workflow history, this marker does not fail replay when Workflow code does not emit it.

Reviewer: ```suggestion
Deprecated patches bridge between `v2` and `vFinal` (the end result).
They work similarly to regular patches by recording a marker in the Workflow history.
This marker does not fail replay when Workflow code does not emit it.
```
